src/Java.Interop.Tools.Cecil/Java.Interop.Tools.Cecil/TypeDefinitionRocks.cs

```diff
@@ -57,8 +55,7 @@ public static bool IsSubclassOf (this TypeDefinition type, string typeName)
 
 		public static bool ImplementsInterface (this TypeDefinition type, string interfaceName)
 		{
-			return type.GetTypeAndBaseTypes ().Any (t => t.HasInterfaces &&
-					t.Interfaces.Any (i => i.InterfaceType.FullName == interfaceName));
+			return type.GetTypeAndBaseTypes ().Any (t => t.Interfaces.Any (i => i.InterfaceType.FullName == interfaceName));
```

Comment:
If we're shooting for performance, this potentially creates a _ton_ of lambda captures.

Reply:
Yeah, I should change this to a foreach loop.